src/helper.cc

```diff
@@ -208,6 +208,8 @@ helperOpenServers(const Helper::Client::Pointer &hlp)
     int wfd;
     void * hIpc;
     wordlist *w;
+    // Helps reducing diff. TODO: remove
+    const auto hlp = this;
```

Comment:
Porting should not be an issue as none of these refactors qualify for backport.

More importantly IMO, is that both of these `openSessions()` methods need a refactor to be better C++ logic and de-duplicate a lot of their actions. Authors choice of scope on whether to include that in-scope - it is clearly related to this PR change, but is NOT required to be done urgently.

Reply:
> More importantly IMO, is that both of these `openSessions()` methods need a refactor to be better C++ logic and de-duplicate a lot of their actions. Authors choice of scope on whether to include that in-scope

Agreed.

> Porting should not be an issue as none of these refactors qualify for backport.

Distinguishing genuine changes from noise becomes a porting issue when the targeted branch has also modified the same code. This may not matter for backporting to other official Squid branch, but porting activities are not limited to those use cases.